### ISSUE TITLE
ESLint support inside of template dynamic expressions

### ISSUE THREAD

**Issue by Ericlm** [vuejs/eslint-plugin-vue#2480]
**Tell us about your environment**

- **ESLint version:** 9.4.0
- **eslint-plugin-vue version:** 9.26.0
- **Vue version:** 3.4.29
- **Node version:** 22.2.0

**The problem you want to solve.**
Dynamic expressions inside of the template part of Vue components are not linted; for example, the following code does not show any linting error related to `no-console` for example:
```vue
<div @click="() => console.log('click triggered')" />
```
It's also a problem when using other ESLint plugins, like stylistic, because conditions like following are not reported/fixed:
```vue
<div v-if="condition1&&condition2" /> // No space before/after operator
```
**Your take on the correct solution to problem.**
I think it's not an easy problem to solve 😅; however I can point out two things:
- TS errors are correctly reported (thanks to `vue-tsc` I guess?)
- Stylistic can report errors within the template (like other plugins), but not within the typescript parts of the template

**Additional context**
We recently switched from prettier to stylistic for more AST-related formatting, and I realized that no linting is happening inside of dynamic expressions.

**Comment by FloEdelmann** [comment]
We have [extension rules](https://eslint.vuejs.org/rules/#extension-rules) that apply ESLint core rules and stylistic rules to JS/TS parts in `<template>`. But they have to be enabled manually.

> TS errors are correctly reported (thanks to `vue-tsc` I guess?)

Yes, that's not ESLint but vue-tsc.

> Stylistic can report errors within the template (like other plugins), but not within the typescript parts of the template

If Stylistic with TypeScript support is installed, the extension rules should also work in TS parts in `<template>`.

**Comment by Ericlm** [comment]
Alright, got it! I ignored that the plugin had all these extension rules. Thank you very much!

PS: as developers always want better tooling even when they have fantastic one, would it be possible to enable/disable the rules based on other plugins configs in the future ?

**Comment by FloEdelmann** [comment]
> would it be possible to enable/disable the rules based on other plugins configs in the future

Well, certainly, but this is out of scope for this repo, and rather should be a separate tool or part of those other configs.

I actually use something like this in https://github.com/OpenLightingProject/open-fixture-library/blob/master/.eslintrc.cjs:

```js
for (const ruleName of vueCoreExtensionRules) {
  if (ruleName in myOwnRuleConfig) {
    myOwnRuleConfig[`vue/${ruleName}`] = myOwnRuleConfig[ruleName];
  }
}
```
